brave/brave-core
# Grant check is now performed when Rewards is enabled and panel is opened

Fixes brave/brave-browser#1798

Per issue, a grant check is already performed when a wallet is created either from the panel or the Optin screen

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:

1. Start Brave with clean profile using `--vmodule=*rewards*=5`
2. Click BAT icon
3. Verify that grant check is not performed (view in terminal)
4. Create wallet from panel
5. Verify that grant check is performed (view in terminal)
6. Close panel and click panel again verifying that grant check is performed every time panel is opened (view in terminal)
7. Claim grant from Rewards page
8. Click BAT icon and verify that grant check results in 404 for payment id every time panel is opened after accepting grant.


## Reviewer Checklist:
